### ISSUE TITLE
'sqlite3 module is required for the SqliteCache' - possibly need to know how to disable zeep cacheing

### ISSUE THREAD

**Issue by mikeatkinsonthesecond** [googleads/googleads-python-lib#326]
Hi,

The Ad Manager API team have suggested I post this here, apologies if it's a stupid question!

I'm following the Google Ad Manager API getting started guide, here - https://developers.google.com/ad-manager/api/start

My problem appears to be the same as openned by chandni28p here - https://github.com/googleads/googleads-python-lib/issues/309

He says "Followed the steps mentioned to change caching mechanism here and the issue got resolved. https://github.com/googleads/googleads-python-lib"

It's not clear to me how to follow those steps at that link - ie presumably I need to disable zeep cacheing, but I can't find the correct syntax for this. Can you shed any light, please?

Thanks.

Example output below:

[root@Zen numberwang]# python test.py
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    network_service = client.GetService('NetworkService', version='v201808')
  File "/usr/local/lib/python2.7/site-packages/googleads/ad_manager.py", line 295, in GetService
    cache=self.cache)
  File "/usr/local/lib/python2.7/site-packages/googleads/common.py", line 1152, in __init__
    transport = _ZeepProxyTransport(timeout, proxy_config, cache)
  File "/usr/local/lib/python2.7/site-packages/googleads/common.py", line 666, in __init__
    cache = zeep.cache.SqliteCache()
  File "/usr/local/lib/python2.7/site-packages/zeep/cache.py", line 67, in __init__
    raise RuntimeError("sqlite3 module is required for the SqliteCache")
RuntimeError: sqlite3 module is required for the SqliteCache

test.py:
# Import the library.
from googleads import ad_manager

# Initialize a client object, by default uses the credentials in ~/googleads.yaml.
client = ad_manager.AdManagerClient.LoadFromStorage()

# Initialize a service.
network_service = client.GetService('NetworkService', version='v201808')

# Make a request.
current_network = network_service.getCurrentNetwork()
print 'Found network %s (%s)!' % (current_network['displayName'],
                                  current_network['networkCode'])

**Comment by msaniscalchi** [comment]
Hello Mike,

Thanks for the report! Perhaps a [more direct link](https://github.com/googleads/googleads-python-lib#how-can-i-configure-or-disable-caching) would be helpful? To be more clear, disabling the cache when using Zeep (i.e. the default SOAP client) would look like:

```
adwords_client = adwords.AdWordsClient(
  developer_token, oauth2_client, user_agent,
  client_customer_id=client_customer_id,
  cache=googleads.common.ZeepServiceProxy.NO_CACHE
```

My understanding that this step would be required in cases where you're running on a more restricted environment, such as App Engine. Although [sqlite3 is included in the standard library](https://docs.python.org/2/library/sqlite3.html), it is [excluded when running on App Engine](https://github.com/mvantellingen/python-zeep/blob/master/src/zeep/cache.py#L13).

Regards,
Mark

**Comment by msaniscalchi** [comment]
> I'm sorry to ask such a basic question, but is there a way to call AdManagerClient.LoadFromStorage() and indicate I want to disable Zeep cacheing? Or can I setup my googleads.yaml to indicate I want to disable Zeep cacheing (I've tried loads of variants with no success)?

Not at this time, although it could hypothetically be supported. I think we didn't prioritize this earlier because the functionality wouldn't benefit App Engine users or those on similarly restricted environments, as they wouldn't have been able to open configuration files at the time this was written. In other words, the expectation was that if you were on a restricted environment, you weren't loading from a file.

Instead, you will need to initialize an AdManagerClient directly. This might look like:

```
ad_manager_client = ad_manager.AdManagerClient(oauth2_client, application_name, 
               network_code=network_code, cache=googleads.common.ZeepServiceProxy.NO_CACHE)
```

Regards,
Mark

**Comment by joshuatimothyg** [comment]
We encountered this when we moved to a new server - it looked weird because it was working on one and not on the other - even if the module versions were the same. Turns out our new server was using python 3.7 with the old one using 3.6. Solved the issue by configuring the new server to use 3.6 instead.

Just thought this info might be of use to someone else. Thanks to the team for the wonderful work on the library!
